src/integrate/scarches/script.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import mudata
+import scvi
+from torch.cuda import is_available as cuda_is_available
+try:
+    from torch.backends.mps import is_available as mps_is_available
+except ModuleNotFoundError:
+    # Older pytorch versions
+    # MacOS GPUs
+    def mps_is_available():
+        return False
+
+### VIASH START
+par = {
+    "input": "resources_test/pbmc_1k_protein_v3/pbmc_1k_protein_v3_mms.h5mu",
+    "modality": "rna",
+    "output": "foo.h5mu",
+    # scANVI parameters
+    "unlabeled_category": "Unknown",
+    "labels_key": "leiden",
+    "predicted_labels_key": "predicted_labels",
+    # Other
+    "obsm_output": "X_integrated_scanvi",
+    "var_input": None,
+    "obs_batch": "sample_id",
+    "input_layer": None,
+    "reduce_lr_on_plateau": True,
+    "lr_factor": 0.6,
+    "lr_patience": 30,
+    "early_stopping": True,
+    "early_stopping_monitor": "elbo_validation",
+    "early_stopping_patience": 45,
+    "early_stopping_min_delta": 0,
+    "max_epochs": 500}
+### VIASH END
+
+def _setup_logger():
+    from sys import stdout
+
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    console_handler = logging.StreamHandler(stdout)
+    logFormatter = logging.Formatter("%(asctime)s %(levelname)-8s %(message)s")
+    console_handler.setFormatter(logFormatter)
+    logger.addHandler(console_handler)
+
+    return logger
+
+PLAN_KWARGS = {
+    "reduce_lr_on_plateau": par['reduce_lr_on_plateau'],
+    "lr_patience": par['lr_patience'],
+    "lr_factor": par['lr_factor'],
+}
+
+
+def map_to_existing_reference(adata_query, model_path, check_val_every_n_epoch=1) -> np.ndarray:
+    """
+    A function to map the query data to the reference atlas
+
+    Input:
+        * adata_query: An AnnData object with the query
+        * model_path: The reference model directory
+
+    Output:
+        A numpy.ndarray containing the embedding coordinates of the query data in the HLCA latent space.
+        If `output_model` is set to True, then the trained reference model is also output
+
+    """
+    print("SCVI-tools version:", scvi.__version__)
+
```

Comment:
Need to load ref model here first to check fields that are required and maybe also model that is required. Then format query data correctly according to ref model

Can we load the model without knowing the model class?

Reply:
> Can we load the model without knowing the model class?
Yes we can, did it
> Need to load ref model here first to check fields that are required and maybe also model that is required. Then format query data correctly according to ref model
Doesn't `prepare_query_anndata()` do it?

Reply:
Good point... does it also do this for layer specification though?